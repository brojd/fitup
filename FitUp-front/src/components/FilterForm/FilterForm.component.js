import React, { PropTypes } from 'react';
import classNames from 'classnames';
import styles from './FilterForm.stylesheet.css';
import CitySelect from '../CitySelect/CitySelect.component';
import SearchBar from '../SearchBar/SearchBar.component';
import AdvancedInput from '../AdvancedInput/AdvancedInput.component';
import Button from '../Button/Button.component';

class FilterForm extends React.Component {
  constructor() {
    super();
    this._handleCityChange = this._handleCityChange.bind(this);
    this._handleClassesChange = this._handleClassesChange.bind(this);
    this._handleGymChange = this._handleGymChange.bind(this);
    this._handleTrainerChange = this._handleTrainerChange.bind(this);
    this._toggleAdvancedSearch = this._toggleAdvancedSearch.bind(this);
    this._handleInputFocus = this._handleInputFocus.bind(this);
    this._handleInputBlur = this._handleInputBlur.bind(this);
    this.state = {
      textInputFocused: true,
      classes: [],
      gym: '',
      trainer: '',
      advancedSearch: false,
      gymsInCity: [],
      trainersInCity: [],
      classesInCity: [],
      overflowVisible: false
    };
  }
  _handleCityChange(city) {
    let trainersInCity = this.props.trainers.filter((trainer) => {
      let currGym = this.props.gyms.find((gym) => gym._id === trainer.gymId);
      return currGym.address.city === city;
    });
    let classesInCity = this.props.classes.filter((oneClasses) => {
      let currGym = this.props.gyms.find((gym) => gym._id === oneClasses.gymId);
      return currGym.address.city === city;
    });
    this.setState({
      city: city,
      gymsInCity: this.props.gyms.filter((gym) => gym.address.city.toLowerCase() === city.toLowerCase()),
      trainersInCity: trainersInCity,
      classesInCity: classesInCity
    });
    this.props.onUserInput(this.state.classes, city, this.state.gym, this.state.trainer);
    return false;
  }
  _handleClassesChange(classes) {
    this.setState({
      classes: classes
    });
    this.props.onUserInput(classes, this.state.city, this.state.gym, this.state.trainer);
    return false;
  }
  _handleGymChange(gym) {
    this.setState({
      gym: gym
    });
    this.props.onUserInput(this.state.classes, this.state.city, gym, this.state.trainer);
    return false;
  }
  _handleTrainerChange(trainer) {
    this.setState({
      trainer: trainer
    });
    this.props.onUserInput(this.state.classes, this.state.city, this.state.gym, trainer);
    return false;
  }
  _toggleAdvancedSearch() {
    this.setState({
      clickedAdvanced: true,
      advancedSearch: !this.state.advancedSearch
    });
  }
  _handleInputFocus() {
    this.setState({ overflowVisible: true });
  }
  _handleInputBlur() {
    this.setState({ overflowVisible: false });
  }
  componentDidMount() {
    let trainersInCity = this.props.trainers.filter((trainer) => {
      let currGym = this.props.gyms.find((gym) => gym._id === trainer.gymId);
      return currGym.address.city === this.props.userLocation;
    });
    let classesInCity = this.props.classes.filter((oneClasses) => {
      let currGym = this.props.gyms.find((gym) => gym._id === oneClasses.gymId);
      return currGym.address.city === this.props.userLocation;
    });
    this.setState({
      city: this.props.userLocation,
      gymsInCity:  this.props.gyms.filter(
        (gym) => gym.address.city.toLowerCase() === this.props.userLocation.toLowerCase()
      ),
      trainersInCity: trainersInCity,
      classesInCity: classesInCity
    });
  }
  render() {
    return (
      <div className={styles.formWrapper}>
        <form className={classNames('uk-width-4-10 uk-form-icon uk-align-center', styles.filterForm,
                        {'rolled': !this.state.advancedSearch && this.state.clickedAdvanced},
                        {'unrolled': this.state.advancedSearch && this.state.clickedAdvanced},
                        {'visible': this.state.overflowVisible},
                        {'hidden': !this.state.overflowVisible })}>
          <CitySelect onChange={this._handleCityChange}
                      gyms={this.props.gyms}
                      userLocation={this.props.userLocation} />
          {this.state.advancedSearch ?
            <Button text='Hide'
                    type='button'
                    className={classNames(styles.advancedButton)}
                    onClick={this._toggleAdvancedSearch} /> :
            <Button text='Advanced'
                    type='button'
                    className={classNames(styles.advancedButton)}
                    onClick={this._toggleAdvancedSearch} />
          }
          <SearchBar
            placeholder='Search for your classes...'
            onChange={this._handleClassesChange}
            onFocus={this._handleInputFocus}
            onBlur={this._handleInputBlur}
            autoFocus={false}
            items={this.state.classesInCity}
          />
          <div className={classNames(
            'uk-width-9-10 uk-align-center',
            styles.advancedInputWrapper,
            {'uk-invisible': !this.state.advancedSearch && this.state.overflowVisible}
            )}>
            <AdvancedInput placeholder='Search by gyms...'
                           onChange={this._handleGymChange}
                           onFocus={this._handleInputFocus}
                           onBlur={this._handleInputBlur}
                           items={this.state.gymsInCity} />
            <AdvancedInput placeholder='Search by trainers...'
                           onChange={this._handleTrainerChange}
                           onFocus={this._handleInputFocus}
                           onBlur={this._handleInputBlur}
                           items={this.state.trainersInCity}
                           className={styles.trainerSearchBar}/>
          </div>
        </form>
      </div>
    );
  }
}

FilterForm.propTypes = {
  onUserInput: PropTypes.func,
  gyms: PropTypes.array,
  classes: PropTypes.array,
  trainers: PropTypes.array,
  userLocation: PropTypes.string
};

FilterForm.defaultProps = {
  gyms: []
};

export default FilterForm;
