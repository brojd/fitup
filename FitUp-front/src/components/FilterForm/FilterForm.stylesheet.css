.formWrapper {
  height: 180px; }

.filterForm {
  height: 78px;
  margin-top: -120px;
  background: rgba(255, 255, 255, 0.4);
  padding: 21px;
  overflow-y: hidden;
  border: 1px solid rgba(167, 0, 0, 0.26);
  border-radius: 2px; }

.advancedButton {
  background: linear-gradient(to bottom right, #482a16, #d0a017) !important;
  letter-spacing: 0.3px;
  color: #f5f5f5 !important;
  font-size: 10px !important;
  position: absolute;
  right: 21px;
  top: 42px;
  z-index: 4;
  width: 16%;
  min-height: 17px !important;
  height: 16px;
  line-height: 16px !important; }

/*# sourceMappingURL=FilterForm.stylesheet.css.map */
