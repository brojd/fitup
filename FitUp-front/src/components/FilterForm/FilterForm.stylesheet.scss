@import "../../../sass_variables/colors";
@import "../../../sass_variables/widths";

.formWrapper {
  height: 180px;
}

.filterForm {
  height: 78px;
  margin-top: -120px;
  background: rgba($white, 0.4);
  padding: 3 * $whitespace;
  overflow-y: hidden;
  border: 1px solid rgba(167, 0, 0, 0.26);
  border-radius: 2px;
}

.advancedButton {
  background: linear-gradient(to bottom right, rgb(72, 42, 22), rgb(208, 160, 23)) !important;
  letter-spacing: 0.3px;
  color: #f5f5f5 !important;
  font-size: 10px !important;
  position: absolute;
  right: 3 * $whitespace;
  top: 28px + (2 * $whitespace);
  z-index: 4;
  width: 16%;
  min-height: 17px !important;
  height: 16px;
  line-height: 16px !important;
}
