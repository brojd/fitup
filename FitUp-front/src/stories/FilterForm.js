import React from 'react';
import { storiesOf} from '@kadira/storybook';
import FilterForm from '../components/FilterForm/FilterForm.component';

storiesOf('FilterForm', module)
  .add('filter form', () => (
    <div style={{margin: '200px 0 0 0'}}>
      <FilterForm />
    </div>
  ));

