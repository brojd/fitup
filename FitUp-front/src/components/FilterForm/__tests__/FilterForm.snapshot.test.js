import React from 'react';
import FilterForm from '../FilterForm.component';
import renderer from 'react-test-renderer';

jest.mock('react-select');

test('FilterForm renders correctly excluding 3rd party react-select', () => {
  
  const gyms = [
    {
      '_id': 1,
      'name': 'Fitness Academy',
      'address': {'city': 'Wrocław', 'postal_code': '50-001', 'street':'Długa', 'number':3},
      'logo_url': 'https://pbs.twimg.com/profile_images/378800000513930765/1518318ade4947d639fca13c47b457f6.jpeg',
      'photo_urls': [
        'https://www.thegymgroup.com/fileadmin/uploads/gym/Photos/Gym_Leicester/Leicester14.jpg'
      ],
      'pricing': [
        {'name':'1 workout', 'price':15}
      ],
      'cards_accepted': [
        {'name':'Multisport', 'accepted':true}, {'name':'FitProfit', 'accepted':false}
      ]
    },
    {
      '_id': 2,
      'name': 'Fitness World',
      'address': {'city': 'Wrocław', 'postal_code': '50-001', 'street':'Olszowa', 'number':3},
      'logo_url': 'http://www.feriogaj.com/wp-content/uploads/fitnes-world.png',
      'photo_urls': [
        'http://www.fru-fitness.pl/content/57/0/1370494719_6ca505f63f8064d96281ac502dd903f2.jpg'
      ],
      'pricing': [
        {'name':'1 workout', 'price':15}
      ],
      'cards_accepted': [
        {'name':'Multisport', 'accepted':false}
      ]
    }
  ];
  const trainers = [
    {
      'name': 'Sabina',
      'surname': 'Nowak',
      'about_me': 'Lorem ipsum dolor sit amet',
      'specialties': {
        'zumba': false,
        'yoga': false,
        'crossfit': false,
        'TBC': false,
        'pilates': true
      },
      'photo_url': 'http://muzeumpmw.pl/wp-content/uploads/2012/09/wpid-trener-personalny-dobrym-wsparciem-w-czasie-odchudzania-081d858b1d0db5ecb5c2a450ef04253c.jpg',
      'gymId': 1
    },
    {
      'name': 'Peter',
      'surname': 'Nowicki',
      'about_me': 'Lorem ipsum dolor sit amet',
      'specialties': {
        'zumba': true,
        'yoga': false,
        'crossfit': false,
        'TBC': false,
        'pilates': true
      },
      'photo_url': 'http://www.treningi-personalne.com/images/Jakub-Kulej-trener-personalny-500.jpg',
      'gymId': 1
    },
    {
      'name': 'Konrad',
      'surname': 'Okowski',
      'about_me': 'Lorem ipsum dolor sit amet',
      'specialties': {
        'zumba': true,
        'yoga': false,
        'crossfit': false,
        'TBC': true,
        'pilates': false
      },
      'photo_url': 'http://www.znanytrener.pl/wp-content/uploads/image/2011/09/pawel-247x300.jpg',
      'gymId': 2
    }
  ];
  const classes = [
    {
      'name': 'yoga',
      'description': 'string',
      'gymId': 1
    },
    {
      'name': 'zumba',
      'description': 'string',
      'gymId': 2
    },
    {
      'name': 'pilates',
      'description': 'string',
      'gymId': 2
    }
  ];
  const component = renderer.create(
    <FilterForm gyms={gyms} trainers={trainers} classes={classes} onUserInput={jest.fn()} userLocation='Wrocław' />
  );
  let tree = component.toJSON();
  
  expect(tree).toMatchSnapshot();
  
});
