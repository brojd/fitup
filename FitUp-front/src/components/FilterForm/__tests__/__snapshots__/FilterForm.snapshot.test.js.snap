exports[`test FilterForm renders correctly excluding 3rd party react-select 1`] = `
<div
  className="formWrapper">
  <form
    className="uk-width-4-10 uk-form-icon uk-align-center filterForm hidden">
    <select
      className="uk-float-right citySelect"
      defaultValue="Wrocław"
      onChange={[Function]}>
      <option
        className="cityOption"
        value="Wrocław">
        Wrocław
      </option>
    </select>
    <button
      autoFocus={false}
      className="uk-button advancedButton"
      disabled={false}
      onClick={[Function]}
      type="button"
      value={undefined}>
      Advanced
    </button>
    <div
      className="uk-width-9-10 uk-align-center advancedInputWrapper">
      <div
        className="advancedInputWrapper" />
      <div
        className="advancedInputWrapper" />
    </div>
  </form>
</div>
`;
